### PR TITLE
Add quickstart template to create Azure DB for MySQL flexible server

### PR DESCRIPTION
**Pull request by shreyaaithal**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by shreyaaithal** [comment]
#sign-off

File: quickstarts/microsoft.dbformysql/create-flexible-server-public-access/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+![Cred Scan Check](https://azurequickstartsservice.blob.core.windows.net/badges/quickstarts/microsoft.dbformysql/create-flexible-server-public-access/CredScanResult.svg)
+
+[![Deploy To Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fquickstarts%2Fmicrosoft.dbformysql%2Fcreate-flexible-server-public-access%2Fazuredeploy.json)
+[![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fquickstarts%2Fmicrosoft.dbformysql%2Fcreate-flexible-server-public-access%2Fazuredeploy.json)
+
+This template creates a Azure Database for MySQL - Flexible Server, configures a server-level firewall rule (public access connectivity method) and creates a database within the server.
+
+## Deployment steps
```
Comment (bmoore-msft) [inline quickstarts/microsoft.dbformysql/create-flexible-server-public-access/README.md:17]: remove this section, the steps must be the same for all samples in the repo

File: quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,161 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "administratorLogin": {
```
Comment (bmoore-msft) [inline quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.json:5]: add metadata descriptions for all parameters

File: quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                    "storageSizeGB": "[parameters('storageSizeGB')]",
+                    "iops": "[parameters('storageIops')]",
+                    "autogrow": "[parameters('storageAutogrow')]"
+                },
+                "Backup": {
+                    "backupRetentionDays": "[parameters('backupRetentionDays')]",
+                    "geoRedundantBackup": "[parameters('geoRedundantBackup')]"
+                } 
+            },
+            "tags": "[parameters('tags')]"
```
Comment (bmoore-msft) [inline quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.json:114]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#sort-order-of-properties for the proper sort order of properties

File: quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+                "Backup": {
+                    "backupRetentionDays": "[parameters('backupRetentionDays')]",
+                    "geoRedundantBackup": "[parameters('geoRedundantBackup')]"
+                } 
+            },
+            "tags": "[parameters('tags')]"  
+        },
+        {
+            "condition": "[greater(length(variables('firewallRules')), 0)]",
+            "type": "Microsoft.Resources/deployments",
```
Comment (bmoore-msft) [inline quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.json:118]: why is this nested?

File: quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.parameters.json [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "administratorLogin": {
+            "value": "GEN-UNIQUE"
+        },
+        "administratorLoginPassword": {
+            "value": "GEN-PASSWORD"
+        },
+        "location": {
```
Comment (bmoore-msft) [inline quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.parameters.json:11]: remove the non-GEN parameter values and use defaultValues in the template where possible.

File: quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.parameters.json [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        },
+        "standbyAvailabilityZone": {
+            "value": ""
+        },
+        "firewallRules": {
+            "value": {
+                "rules": [
+                    {
+                        "name" : "AllowMyIP",
+                        "startIPAddress" : "59.97.41.110",
```
Comment (bmoore-msft) [inline quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.parameters.json:52]: What are these ip addresses?

**Review by bmoore-msft — CHANGES_REQUESTED** [review]
Also check the pipeline results for other changes that need to be made...

File: quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.json (hunk 4 above)
Comment (shreyaaithal) [reply quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.json:118]: For adding multiple firewall rules

File: quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.parameters.json (hunk 6 above)
Comment (shreyaaithal) [reply quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.parameters.json:52]: Sample IP addresses. Will replace them with generic values.

**Comment by shreyaaithal** [comment]
@bmoore-msft - Build is failing due to the sole reason that we are capacity constrained in the "westus" region where this arm template is being deployed for testing. Hardcoding the region parameter is causing best practice violation.
Please suggest the way forward.

**Comment by bmoore-msft** [comment]
Add the location parameter to the parameters file and you can select another region that way...

**Comment by shreyaaithal** [comment]
#sign-off

File: quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.json (hunk 4 above)
Comment (bmoore-msft) [reply quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.json:164]: you don't need to nest the deployment for that - you can just use a copy loop on the resource itself.

File: quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.parameters.json (hunk 6 above)
Comment (bmoore-msft) [reply quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.parameters.json]: use something that will make it obvious they are "placeholder" and not expose the resource to an unknown IP... depends on what you support, could be 0.0.0.0, 10. space, 169. etc

**Comment by shreyaaithal** [comment]
@bmoore-msft - I have made all the corrections. Can you please review and merge this PR?

**Comment by bmoore-msft** [comment]
see comments above from the previous review

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback.

**Pull request closed without merging**